### ISSUE TITLE
[Bug]: pipeline.load("./pipeline_storage") is looking for docstore.json

### ISSUE THREAD

**Issue by tmceld** [run-llama/llama_index#10244]
### Bug Description

I am getting an error when trying to `new_pipeline.load("./pipeline_storage")` as per [this example](https://docs.llamaindex.ai/en/stable/module_guides/loading/ingestion_pipeline/root.html#caching)

the error is:
`FileNotFoundError: [Errno 2] No such file or directory: '/Users/toast/Developer/LLM/guidelines/pipeline_storage/docstore.json'`

### Version

0.9.35

### Steps to Reproduce

My code is:
```python
from llama_index.storage.docstore import SimpleDocumentStore
from llama_index.storage import StorageContext

# Create a docstore
docstore = SimpleDocumentStore()

llm = Ollama(model="llama2", request_timeout=300.0)
service_context = ServiceContext.from_defaults(
llm=llm, chunk_size=256,
embed_model="local")
set_global_service_context(service_context)
documents = SimpleDirectoryReader("docs").load_data()
# Add documents to the docstore
docstore.add_documents(documents)

... other code here ...

from llama_index.ingestion import IngestionPipeline
from multiprocessing import set_start_method

print('getting IngestionPipeline')
pipeline = IngestionPipeline(transformations=transformations)

print('running pipeline')
set_start_method("spawn", force=True)  # it hangs without this line
uber_nodes = pipeline.run(documents=documents)

# save
pipeline.persist("./pipeline_storage")

new_pipeline = IngestionPipeline(transformations=transformations)
new_pipeline.load("./pipeline_storage")
```

Now, i had a chat with kapa.ai on the discord, which pointed out that i don't have a docstore, which is fair.  so it recommended the following code:

```python
from llama_index.storage.docstore import SimpleDocumentStore 
from llama_index.storage import StorageContext
# Create a docstore
docstore = SimpleDocumentStore() 
llm = Ollama(model="llama2", request_timeout=300.0) 
service_context = ServiceContext.from_defaults( 
											   llm=llm, 
											   chunk_size=256,
											   embed_model="local"
											   ) 
										set_global_service_context(service_context) documents = SimpleDirectoryReader("docs").load_data()

# Add documents to the docstore

docstore.add_documents(documents)

# Create a storage context with the docstore

storage_context = StorageContext.from_defaults(docstore=docstore) 

... other code here ... 

from llama_index.ingestion import IngestionPipeline 
from multiprocessing import set_start_method 

print('getting IngestionPipeline')

# Pass the storage context to the IngestionPipeline

pipeline = IngestionPipeline(transformations=transformations, storage_context=storage_context) print('running pipeline') set_start_method("spawn", force=True) # it hangs without this line uber_nodes = pipeline.run(documents=documents)

# save

pipeline.persist("./pipeline_storage") 
```

the problem here is that IngestionPipeline does not in fact take a storage_context.

in summary:
1. There may be a bug if it doesn't find a docstore
2. I am perfectly happy to create a docstore, but how do i persist that?

Many thanks

**Comment by tmceld** [comment]
This is good, i can persist the docstore to that location, TY.  @logan-markewich , over on discord also suggested:
`pipeline = IngestionPipeline(..., docstore=SimpleDocumentStore())` which i will look into.  Thanks.
